### PR TITLE
Add possibility to add custom init-container and lifecycle for master-container

### PR DESCRIPTION
**Pull request by Anarhyst266**
#### What this PR does / why we need it:
We need to start custom init-containers for dynamically specifying environment variables to JCasC, like CASC_VAULT_TOKEN
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/jenkins/README.md [hunk 1]
```diff
@@ -102,6 +102,8 @@ The following tables list the configurable parameters of the Jenkins chart and t
 | `Master.CustomConfigMap`          | Deprecated: Use a custom ConfigMap               | `false`                                                                      |
 | `Master.AdditionalConfig`         | Deprecated: Add additional config files         | `{}`
 | `Master.JenkinsUriPrefix`         | Root Uri Jenkins will be served on         | Not set
+| `Master.CustomInitContainers`     | Custom init-container specification in raw-yaml format | Not set
+| `Master.Lifetime`                 | Lifetime specification for master-container | Not set
```
Comment (ipleten) [inline stable/jenkins/README.md]: It should be 'Lifecycle'

Comment (Anarhyst266) [reply stable/jenkins/README.md]: It's fixed, thank you!

**Comment by maorfr** [comment]
/assign

hey @Anarhyst266, thanks for contributing!
may i ask what is the problem you are trying to solve?
i am usually more in favor of PRs that expose a feature rather then the option to add more yaml to the values file.

**Comment by Anarhyst266** [comment]
Hello, @maorfr. 
I'm trying to integrate Jenkins with bootsport's kubernetes-vault, so I need to start additional init-container which will get vault-token for JCasC and add lifecycle to main container which will prepare CASC_VAULT_FILE from that token. Also I think that additional init-containers and lifecycle may be needed for some another Jenkins integrations with another tools. 
Thank you.

**Comment by maorfr** [comment]
/ok-to-test

i seem to always be on the losing side of the customInitContainers discussion ;)
can you just change the names to be snakeCase? we are moving towards [naming convention best practices](https://helm.sh/docs/chart_best_practices/#naming-conventions).

**Comment by Anarhyst266** [comment]
Changed :-)

File: stable/jenkins/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: jenkins
 home: https://jenkins.io/
-version: 0.38.0
+version: 0.38.1
```
Comment (maorfr) [inline stable/jenkins/Chart.yaml]: also lets bump this to 0.39.0 please

Comment (maorfr) [reply stable/jenkins/Chart.yaml]: sorry for not saying that at first

Comment (Anarhyst266) [reply stable/jenkins/Chart.yaml]: OK, done

**Comment by maorfr** [comment]
/lgtm

thanks for contributing and congrats on your first contribution!

**Pull request merged**